docs/subnets/cross-subnet-communication.md

```diff
@@ -0,0 +1,72 @@
+# Cross-Subnet Communication
+
+Avalanche Warp Messaging (AWM) enables native cross-Subnet communication and allows [Virtual Machine
+(VM)](./README.md#virtual-machines) developers to utilize any-to-any communication between the
+validators of two Subnets. The content of the messages can contain arbitrary payloads. 
+
+## Use Cases 
+
+Use cases for AWM may include but is not limited to:
+
+- Orcale Networks: Connecting a Subnet to an oracle network is a costly process. AWM makes it easy
+  for oracle networks to broadcast their data from their origin chain to other Subnets.
+- Token transfers between Subnets 
+- State Sharding between multiple Subnets
+
+## Elements of Cross-Subnet Communication
+
+The communication consists of the following four steps:
+
+![image showing four steps of cross-Subnet communication: Signing, aggregation, Delivery and Verification](/img/cross-subnet-communication.png)
+
+### 1.) Signing Messages on the Origin Subnet
+
+AWM is a low-level messaging protocol. Any type of data encoded in an array of bytes can be included
+in the message sent to another Subnet. AWM uses the [BLS signature
+scheme](https://crypto.stanford.edu/~dabo/pubs/papers/BLSmultisig.html), which allows message
+recipients to verify the authenticity of these messages. Therefore, every validator on the Avalanche
+network holds a BLS key pair, consisting of a private key for signing messages and a public key that
+others can use to verify the signature.
+
+### 2.) Signature Aggregation on the Origin Subnet
+
+If the validator set of a Subnet is very large, this would result in the Subnet's validators sending
+many signatures between them. One of the powerful features of BLS is the ability to aggregate many
+signatures of different signers in a single multi-signature. Therefore, validators of one Subnet can
+now individually sign a message and these signatures are then aggregated into a short
+multi-signature that can be quickly verified.
+
+### 3.) Delivery of Messages to the Destination Subnet
+
+Messages are delivered directly from the validators of one Subnet to the validators of the other
+Subnets. The messages do not pass through a central protocol or trusted entity, and there is no
```

Comment:
```
Messages are delivered directly from the validators of one Subnet to the validators of the other
Subnets
```
This seems to contradict the below statement:
```
It is up to the Subnets and their users to determine how they want to transport data from the
validators of the origin Subnet to the validators of the destination Subnet and what guarantees they
want to provide for the transport.
```
I think that the first sentence is a bit confusing / misleading. The second paragraph is exactly correct.